### DISCUSSION TITLE
help to understand how does this plugin work in a concrete example

### DISCUSSION THREAD

**Discussion by cakkypamucm** [ismay/stylelint-no-unsupported-browser-features#271] [Q&A] [answered]
First of all, thank you, @ismay, for [quick informative answer](https://github.com/ismay/stylelint-no-unsupported-browser-features/issues/270#issuecomment-1509885997)!
Yes, my bad - it's not a bug of `ignorePartialSupport` option.  

But I continue to think there is a bug (or, of course, maybe I don't understand how does this plugin work)  
So, let me try to explain again, please.

- I have stylelintrc config:
```
{
  "plugins": [
    "stylelint-no-unsupported-browser-features"
  ],
  "rules": {
    "plugin/no-unsupported-browser-features": [true]
  }
}
```
- and browserslistrc config:
```
> 1%
last 2 versions and not dead
not op_mini all
```
As you can [see](https://browsersl.ist/#q=%3E+1%25%2Clast+2+versions+and+not+dead%2Cnot+op_mini+all), there is `Samsung Internet` browser in the list  

- I execute `stylelint --fix` and among checking files there is a one (with `css` extension) with line
`height: 100dvh;`
As you can [see](https://caniuse.com/?search=dvh), `Samsung Internet` is not supported `dvh`

> This plugin checks if the CSS you're using is supported by the browsers you're targeting

In sum: some my targeting browsers are supported `dvh` and some are not (like `Samsung Internet`)  
And I expect warning about `Samsung Internet` and `dvh` - but there is no warning.  
Why?

**Comment by ghost** [comment, marked as answer]
So there are a lot of things that could affect this not working as expected, off the top of my head:

- Your config seems all right, though I'd recommend setting the severity to warning as recommended in the readme.
- Ensure that your browserslist has been configured correctly by running `npx browserslist` in your project folder. It will show you the list of browsers that match your query. Check if the browsers you expect it to return are in there.
- Ensure that the file you're linting is being linted correctly by stylelint. Try to trigger a warning or error and see if that works.
- Ensure that stylelint-no-unsupported-browser-features has been set up correctly by triggering an error or warning with some other property.

If that is all correct it is possible that doiuse does not detect use of dvh. This plugin uses doiuse to detect browser support, so if doiuse doesn't detect it there's nothing this plugin can do. So if the above is all correct and you're still not getting a warning from this plugin you should double check that doiuse supports it: https://github.com/anandthakker/doiuse. Run doiuse directly against the file with the `dvh` rule and see if you get a warning. If doiuse warns about it, this plugin should as well.

**Reply by cakkypamucm** [reply]
Thanks for support!
Sorry for my so slow reaction.
